zhmccli/_cmd_cpc.py

```diff
@@ -1093,16 +1116,27 @@ def cmd_cpc_upgrade(cmd_ctx, cpc_name, options):
             "the Web Services API".format(v=hmc_version),
             cmd_ctx.error_format)
 
-    click.echo("Upgrading the SE of CPC {c} to bundle level {bl} and waiting "
+    fw_str = "FTP server {fs!r}".format(fs=ftp_host) if ftp_host else \
+        "the IBM support site"
+    click.echo("Upgrading the SE of CPC {c} to bundle level {bl} with "
+               "firmware retrieval from {fw} and waiting "
                "for completion (timeout: {t} s)".
-               format(c=cpc_name, bl=bundle_level, t=timeout))
+               format(c=cpc_name, bl=bundle_level, fw=fw_str, t=timeout))
+
+    kwargs = dict(
```

Comment:
code duplication? could the two ifs go into a utility method?

Reply:
I don't see any two if's near the location you commented on. The one if near the location occurs only once in the source file. Can you please elaborate?

Reply:
Well, it is "duplicated code" if there is a realistic chance that future changes would affect all those "copies".
In the end, it is your call to make ;-)